syz-manager/rpc.go

```diff
@@ -101,16 +94,18 @@ type BugFrames struct {
 type RPCManagerView interface {
 	currentBugFrames() BugFrames
 	machineChecked(features flatrpc.Feature, enabledSyscalls map[*prog.Syscall]bool)
-	getFuzzer() *fuzzer.Fuzzer
 }
 
-func startRPCServer(mgr *Manager) (*RPCServer, error) {
+func startRPCServer(mgr *Manager, source queue.Source) (*RPCServer, error) {
+	var checkSource queue.DynamicSource
 	serv := &RPCServer{
-		mgr:       mgr,
-		cfg:       mgr.cfg,
-		target:    mgr.target,
-		checker:   vminfo.New(mgr.cfg),
-		statExecs: mgr.statExecs,
+		mgr:           mgr,
+		cfg:           mgr.cfg,
+		target:        mgr.target,
+		checker:       vminfo.New(mgr.cfg),
+		checkerSource: &checkSource,
+		execSource:    queue.Retry(queue.Order(&checkSource, source)),
```

Comment:
Humm... I assumed it will be a single dynamic source instead of order + 2 dynamic sources + dynamic source supporting nil value.
I.e. RPCServer contains DynamicSource, we first store checker in it, then when the fuzzer is created we store fuzzer in it. In this scheme we can assume DynamicSource internal pointer is never nil.

Reply:
I think it's also a question of the interaction between `manager.go` and `rpc.go` -- what does what. Now it's something like:
* `rpc.go` performs a machine check.
  * It happens asynchronously w.r.t. `manager.go`
* Once machine check is done, `manager.go` creates a `Fuzzer` object and connects it to the `queue.Source`.
  * It's transparent for `rpc.go`; it does not explicitly await it.

We could just add `SetSource()` to `rpc.go` and call it after the check is done (and it will work), but I'm afraid it will make the interaction even more brittle and hard to understand. It's two objects actively interacting with each other.

`rpc.go`'s interface could become more passive:
```
MachineCheck() CheckResult -- blocks until the checking is done.
Run(queue.Source) -- blocks forever.
```

It will actually make much closer to what we now do in `tools/syz-runtest`.

Reply:
I think the interaction will become even clearer if machineChecked will return new Source.
startRPCServer won't need to accept anything. startRPCServer will create one DynamicSource, initialize with checker, then store machineChecked result.

Reply:
Done